datadog_checks_base/datadog_checks/base/checks/openmetrics/mixins.py

```diff
@@ -880,3 +883,49 @@ def _get_bearer_token(self, bearer_token_auth, bearer_token_path):
         except Exception as err:
             self.log.error("Cannot get bearer token from path: %s - error: %s", path, err)
             raise
+
+    def _histogram_convert_values(self, metric_name, converter):
+        def _convert(metric, scraper_config=None):
+            for index, sample in enumerate(metric.samples):
+                val = sample[self.SAMPLE_VALUE]
+                if not self._is_value_valid(val):
+                    self.log.debug("Metric value is not supported for metric {}".format(sample[self.SAMPLE_NAME]))
+                    continue
+                if sample[self.SAMPLE_NAME].endswith("_sum"):
+                    lst = list(sample)
+                    lst[self.SAMPLE_VALUE] = converter(float(val))
+                    metric.samples[index] = tuple(lst)
+                elif sample[self.SAMPLE_NAME].endswith("_bucket") and "Inf" not in sample[self.SAMPLE_LABELS]["le"]:
+                    sample[self.SAMPLE_LABELS]["le"] = str(converter(float(sample[self.SAMPLE_LABELS]["le"])))
+            self.submit_openmetric(metric_name, metric, scraper_config)
+
+        return _convert
+
+    def _histogram_from_microseconds_to_seconds(self, metric_name):
+        return self._histogram_convert_values(metric_name, lambda v: v / self.MICROS_IN_S)
+
+    def _histogram_from_seconds_to_microseconds(self, metric_name):
+        return self._histogram_convert_values(metric_name, lambda v: v * self.MICROS_IN_S)
+
+    def _summary_convert_values(self, metric_name, converter):
+        def _convert(metric, scraper_config=None):
+            for index, sample in enumerate(metric.samples):
+                val = sample[self.SAMPLE_VALUE]
+                if not self._is_value_valid(val):
+                    self.log.debug("Metric value is not supported for metric {}".format(sample[self.SAMPLE_NAME]))
+                    continue
+                if sample[self.SAMPLE_NAME].endswith("_count"):
+                    continue
+                else:
+                    lst = list(sample)
+                    lst[self.SAMPLE_VALUE] = converter(float(val))
```

Comment:
Why the float cast?

Reply:
Good point, thank you @ofek, the only spot where we need the cast is on the label (https://github.com/DataDog/integrations-core/pull/5364/files/c6bcff03aef5158d5528ef3b0877d375b2df56df#diff-85fd7415439d33e481bee8605a392317R899). I will remove casting in the other two spots.